src/transformers/models/mixtral/modeling_mixtral.py

```diff
@@ -95,21 +100,45 @@ def load_balancing_loss_func(gate_logits: torch.Tensor, num_experts: torch.Tenso
     if gate_logits is None or not isinstance(gate_logits, tuple):
         return 0
 
+    batch_size, sequence_length = attention_mask.shape
```

Comment:
attention mask can be None, let's make sure this is taken into account, I'd rather we don't create / compute everything with a dummy mask if no mask is provided!

Reply:
Actually, I already **checked if attention_mask is None** at where we use the function:
https://github.com/khaimt/transformers/blob/exclude_padding_tokens_in_aux_loss_Mixtral_Moe/src/transformers/models/mixtral/modeling_mixtral.py#L1380

Reply:
I don't think computing more things is better than skipping this!

Reply:
Yes, Let me update the code to skip computing more when attention_mask is None

Reply:
@ArthurZucker I have just updated the code for skipping computing more when attention_mask is None